### PR TITLE
Feature/usecase interfaces

### PR DESCRIPTION
**Pull request by sven1103**

### CODE REVIEW

**Comment by KochTobi** [comment]
Maybe always provide the sample id as first parameter. That way it is more intuitive at what position it is entered.

File: src/main/groovy/life/qbic/portal/sampletracking/app/samples/query/QuerySampleTrackingInfo.groovy [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            output.currentLocation(location)
+        } catch (SampleTrackingQueryException e) {
+            log.error e
+            output.invokeOnError "Could not get current location for sample $sampleId"
+        }
+    }
+
+
+    @Override
+    def availableLocationsForPersonWithEmail(String email, SampleStatusOutput output) {
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/portal/sampletracking/app/samples/query/QuerySampleTrackingInfo.groovy]:
```suggestion
    def availableLocationsForEmail(String email, SampleStatusOutput output) {
```
Persons with is not needed here I think

File: src/main/groovy/life/qbic/portal/sampletracking/app/samples/query/SampleStatusOutput.groovy [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package life.qbic.portal.sampletracking.app.samples.query
+
+import life.qbic.datamodel.samples.Location
+
+interface SampleStatusOutput {
+
+    def availableLocations(List<Location> locations)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/portal/sampletracking/app/samples/query/SampleStatusOutput.groovy]: I do not fully understand this class. It should implemented by the vaadin layout class. I do not understand the method names and where they are to be used in this case.

Maybe rename to `updateAvailableLocations`

**Comment by sven1103** [comment]
@KochTobi thank you !

**Review by Steffengreiner — APPROVED** [review]
Looks fine, Thanks for taking the time to refactor the code!

**Pull request merged**